README.md

```diff
@@ -308,13 +314,179 @@ securely.
+         - <PATH-TO-CONJUR-DATA-DIRECTORY>:/var/lib/postgresql/data
+   ```
+
+   For example:
+
+   ```
+       volumes:
+         - /home/myusername/conjur-quickstart/temp-db-data:/var/lib/postgresql/data
+   ```
+
+1. Start the Conjur OSS environment using persistence:
+
+   - If you had previously been running the Conjur OSS environment,
+     follow the steps outlined above starting with Step 4 of the
+     [Set up a Conjur OSS environment](#set-up-a-conjur-oss-environment)
+     section above.
+   - Otherwise, follow the steps starting with Step 1 of the
+     [Set up a Conjur OSS environment](#set-up-a-conjur-oss-environment)
+     section above.
+
+#### Restarting the Conjur OSS environment using persistence
+
+Once you have set up the Conjur OSS environment to support persistent Conjur
+state, you can restart your environment as follows:
+
+1. Bring the containers down:
+
+   ```
+   docker-compose down
+   ```
+
+   _**NOTE: You must use the `docker-compose down` command here rather than
+   the `docker-compose stop` in order to avoid having stale, ephemeral
+   connection state in the Conjur container. If you use the `docker-compose
+   stop` command here instead, you may see errors as described in the
+   [`Failed to open TCP connection` error for Conjur login](#failed-to-open-tcp-connection-error-for-conjur-login)
+   section below.**_
+
+1. Bring the containers back up:
+
+   ```
+   docker-compose up -d
+   ```
+
+1. Reconnect the Conjur client to the Conjur server. Use the account name
+   that you created in the
+   [Create an admin account](#create-an-admin-account) section above. For
+   example:
+
+   ```
+   docker-compose exec client conjur init -u conjur -a myConjurAccount
+   ```
+
+1. Log in again to Conjur as admin. When prompted for a password, insert the
+   API key stored in the `admin_data` file:
+
+   ```
+   docker-compose exec client conjur authn login -u admin
+   ```
+
+   **Verification**
+   When you successfully log in, the terminal returns:
+   ```
+   Logged in
+   ```
+
+#### Delete the Conjur data directory when done
+
+For added security, remember to delete the data directory that you created
+in Step 1 of the
+[Set up a Conjur OSS environment with persistence](#set-up-a-conjur-oss-environment-with-persistence)
+section above.
+
 ### Next steps
 
 Now that you've got a local Conjur instance running, what can you do with it?
 
 Try some of our [tutorials](https://www.conjur.org/get-started/tutorials/) on
 Conjur.org.
 
+## Troubleshooting
+
+### `Failed to open TCP connection` error for Conjur login
+
+If you are
+[using persistent Conjur configuration](#using-persistent-conjur-configuration),
+and you see the following error when trying to log into Conjur:
+
+```
+error: Failed to open TCP connection to conjur:80 (Connection refused - connect(2) for "conjur" port 80)
+```
+
+Then try the following:
+
+1. Run the following command:
+
+   ```
+   docker logs $conjur | grep "already running"
```

Comment:
`docker` or `docker-compose` here? If it's the former, `$conjur` needs to be explained (or changed). If it's the latter, we should be able to use the compose service name instead for `$conjur`.

Reply:
it seems to me like there is a push to use `docker-compose` throughout, so if we can do that here too, let's

Reply:
I switched this to use docker-compose.